src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -202,160 +208,205 @@ public CassandraStorage(ReaperApplicationConfiguration config, Environment envir
+                      + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")
+              .setConsistencyLevel(ConsistencyLevel.QUORUM);
+      getRepairSchedulePrepStmt
+          = session
+              .prepare("SELECT * FROM repair_schedule_v1 WHERE id = ?")
+              .setConsistencyLevel(ConsistencyLevel.QUORUM);
+      insertRepairScheduleByClusterAndKsPrepStmt = session.prepare(
+          "INSERT INTO repair_schedule_by_cluster_and_keyspace(cluster_name, keyspace_name, repair_schedule_id)"
+              + " VALUES(?, ?, ?)");
+      getRepairScheduleByClusterAndKsPrepStmt = session.prepare(
+          "SELECT repair_schedule_id FROM repair_schedule_by_cluster_and_keyspace "
+              + "WHERE cluster_name = ? and keyspace_name = ?");
+      deleteRepairSchedulePrepStmt = session.prepare("DELETE FROM repair_schedule_v1 WHERE id = ?");
+      deleteRepairScheduleByClusterAndKsPrepStmt = session.prepare(
+          "DELETE FROM repair_schedule_by_cluster_and_keyspace "
+              + "WHERE cluster_name = ? and keyspace_name = ? and repair_schedule_id = ?");
+      prepareLeaderElectionStatements(timeUdf);
+      getRunningReapersCountPrepStmt = session.prepare(SELECT_RUNNING_REAPERS);
+      saveHeartbeatPrepStmt = session
+          .prepare(
+              "INSERT INTO running_reapers(reaper_instance_id, reaper_instance_host, last_heartbeat)"
+                  + " VALUES(?,?," + timeUdf + "(now()))")
+          .setIdempotent(false);
+
+      getSnapshotPrepStmt = session.prepare("SELECT * FROM snapshot WHERE cluster = ? and snapshot_name = ?");
+      deleteSnapshotPrepStmt = session.prepare("DELETE FROM snapshot WHERE cluster = ? and snapshot_name = ?");
+      saveSnapshotPrepStmt = session.prepare(
+              "INSERT INTO snapshot (cluster, snapshot_name, owner, cause, creation_time)"
+                  + " VALUES(?,?,?,?,?)");
+
+      if (0 >= VersionNumber.parse("3.0").compareTo(version)) {
+        try {
+          getRepairSegmentsByRunIdAndStatePrepStmt = session.prepare(
+              "SELECT id,repair_unit_id,segment_id,start_token,end_token,segment_state,coordinator_host,"
+                  + "segment_start_time,segment_end_time,fail_count, token_ranges FROM repair_run "
+                  + "WHERE id = ? AND segment_state = ? ALLOW FILTERING");
+          getRepairSegmentCountByRunIdAndStatePrepStmt = session.prepare(
+              "SELECT count(segment_id) FROM repair_run WHERE id = ? AND segment_state = ? ALLOW FILTERING");
+        } catch (InvalidQueryException ex) {
+          throw new AssertionError(
+              "Failure preparing `SELECT… FROM repair_run WHERE… ALLOW FILTERING` should only happen on Cassandra-2",
+              ex);
+        }
       }
+      prepareMetricStatements();
+    } catch (RuntimeException e) {
+      LOG.error("Failed preparing Cassandra statements", e);
+      throw new ReaperException(e);
```

Comment:
can we take the try-catch out of this PR please. it's generated such a huge diff that it's difficult to see what's actually changed here. i don't think it's specific to this PR either?

Reply:
ok